### PR TITLE
Add file and line

### PR DESCRIPTION
**Pull request by eranra**

### CODE REVIEW

File: log/logger.go [hunk 1]
```diff
@@ -19,6 +21,8 @@ const (
 	TimeStampKey = "ts"
 	ComponentKey = "component"
 	LevelKey     = "level"
+	FileKey      = "file"
```
Comment (alanconway) [inline log/logger.go:24]: Use a single FileLineKey with the format "file:line" - less clutter in the log message and better chance that editors will be able to recognize as a source location.

File: log/logger.go [hunk 2]
```diff
@@ -112,11 +117,16 @@ func (l *Logger) Enabled() bool {
 // log will log the message. It DOES NOT check Enabled() first so that should
 // be checked by it's callers
 func (l *Logger) log(msg string, context map[string]interface{}) {
+	_, file, line, _ := runtime.Caller(3)
+	file = filepath.Base(filepath.Dir(file)) + "/" + filepath.Base(file)
```
Comment (alanconway) [inline log/logger.go:121]: Suggest computing the path like this:
```
func sourcePath(file string) string {
	if wd, err := os.Getwd(); err == nil {
		if rel, err := filepath.Rel(wd, file); err == nil && !strings.HasPrefix(rel, "../") {
			return rel
		}
	}
	return filepath.Join(filepath.Base(filepath.Dir(file)), filepath.Base(file))
}

```
That way, if your working directory is inside the source tree, you'll get a valid relative path that an editor can follow. If you're outside, fall back to your format.

File: log/observer_test.go [hunk 3]
```diff
@@ -127,6 +131,18 @@ func parseEntry(entryMap map[string]interface{}) *observedEntry {
 	result := &observedEntry{}
 	var ok bool
 
+	result.File, ok = m[log.FileKey].(string)
+	delete(m, log.FileKey)
+	if !ok {
```
Comment (alanconway) [inline log/observer_test.go:136]: Make FileLineKey optional, not an error if missing.

**Review by alanconway — CHANGES_REQUESTED** [review]
Useful for devs, but logs are user facing and source locations make no sense to users. Suggest making this optional, enable with env var LOG_DEV=1. (If we add other optional developer-only features in future we can enable them with the same env var.) A couple of other suggestions, but good idea!

**Comment by eranra** [comment]
Moving and fixing all of those in PR #16 https://github.com/ViaQ/logerr/pull/16

**Comment by eranra** [comment]
Closing this PR moving all to #16

**Pull request closed without merging**